meson.build

```diff
@@ -22,7 +22,7 @@ py = py_mod.find_installation('python')
 py_dep = py.dependency()
 tempita = files('generate_pxi.py')
 versioneer = files('generate_version.py')
-
+openmp = dependency('openmp', required: false)
```

Comment:
does this mean that i shouldn't expect this PR to affect perf on mac?

Reply:
If using the standard compiler I don't think so, but you may be able to achieve it if you use a conda supplied installer. 

There are two ways to validate:
  1. When installing with meson add the `--config-settings=compile-args="--verbose"` flag, i.e. `python -m pip install -ve . --no-build-isolation --config-settings=compile-args="--verbose"` and search for the `-fopenmp` flag being passed when linking the groupby lib
  2. grep the groupby lib for any openmp symbols. On linux this would be `objdump --syms build/cp311/pandas/_libs/groupby.cpython-311-x86_64-linux-gnu.so | grep omp_get` with the standard meson build directory + py311. You will need to tweak to your environment, and maybe tweak objdump (not sure how it works on macOS)

Here's the sample output from 2 on my computer:

```sh
objdump --syms build/cp311/pandas/_libs/groupby.cpython-311-x86_64-linux-gnu.so | grep omp_get
0000000000000000       F *UND*	0000000000000000              omp_get_thread_num@OMP_1.0
0000000000000000       F *UND*	0000000000000000              omp_get_num_threads@OMP_1.0
```

The `omp_get` symbols are of interest

Reply:
I guess the second solution above is the only "guaranteed" way of knowing; for option 1 it is possible for the flag to be silently ignored